[BUG] compiling error with DEBUG_EEPROM_READWRITE

### Description

PR https://github.com/MarlinFirmware/Marlin/commit/380c771988ff415d48cc06df74b6c86967087016#diff-095d5ce4ba7d3ee64f2dc65b6f0d0ebc

With defined `DEBUG_EEPROM_READWRITE` becomes compile error.

```
Marlin\src\module\configuration_store.cpp:411:25: error: 'eeprom_index' was not declared in this scope
int &test_index = eeprom_index;
^
```

### Steps to Reproduce

1. actives `#define DEBUG_EEPROM_READWRITE`
2. compile
[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/3002576/Marlin.zip)

--- a/Marlin/src/module/configuration_store.cpp
+++ b/Marlin/src/module/configuration_store.cpp
@@ -397,7 +397,9 @@ void MarlinSettings::postprocess() {
     #define EEPROM_SKIP(VAR) (eeprom_index += sizeof(VAR))
   #endif
 
-  #define EEPROM_START()          int eeprom_index = EEPROM_OFFSET; persistentStore.access_start()
+  int eeprom_index = EEPROM_OFFSET;
+
+  #define EEPROM_START()          persistentStore.access_start()
   #define EEPROM_FINISH()         persistentStore.access_finish()
   #define EEPROM_WRITE(VAR)       do{ persistentStore.write_data(eeprom_index, (uint8_t*)&VAR, sizeof(VAR), &working_crc);              UPDATE_TEST_INDEX(VAR); }while(0)
   #define EEPROM_READ(VAR)        do{ persistentStore.read_data(eeprom_index, (uint8_t*)&VAR, sizeof(VAR), &working_crc, !validating);  UPDATE_TEST_INDEX(VAR); }while(0)

--- a/Marlin/src/module/configuration_store.cpp
+++ b/Marlin/src/module/configuration_store.cpp
@@ -397,8 +397,6 @@ void MarlinSettings::postprocess() {
     #define EEPROM_SKIP(VAR) (eeprom_index += sizeof(VAR))
   #endif
 
-  int eeprom_index = EEPROM_OFFSET;
-
   #define EEPROM_START()          int eeprom_index = EEPROM_OFFSET; persistentStore.access_start()
   #define EEPROM_FINISH()         persistentStore.access_finish()
   #define EEPROM_WRITE(VAR)       do{ persistentStore.write_data(eeprom_index, (uint8_t*)&VAR, sizeof(VAR), &working_crc);              UPDATE_TEST_INDEX(VAR); }while(0)

--- a/Marlin/src/module/configuration_store.cpp
+++ b/Marlin/src/module/configuration_store.cpp
@@ -399,7 +399,7 @@ void MarlinSettings::postprocess() {
 
   int eeprom_index = EEPROM_OFFSET;
 
-  #define EEPROM_START()          persistentStore.access_start()
+  #define EEPROM_START()          int eeprom_index = EEPROM_OFFSET; persistentStore.access_start()
   #define EEPROM_FINISH()         persistentStore.access_finish()
   #define EEPROM_WRITE(VAR)       do{ persistentStore.write_data(eeprom_index, (uint8_t*)&VAR, sizeof(VAR), &working_crc);              UPDATE_TEST_INDEX(VAR); }while(0)
   #define EEPROM_READ(VAR)        do{ persistentStore.read_data(eeprom_index, (uint8_t*)&VAR, sizeof(VAR), &working_crc, !validating);  UPDATE_TEST_INDEX(VAR); }while(0)
@@ -410,7 +410,7 @@ void MarlinSettings::postprocess() {
     #if WORD_PADDED_EEPROM
       int test_index;
     #else
-      int &test_index = eeprom_index;
+      #define test_index eeprom_index
     #endif
     #define _FIELD_TEST(FIELD) \
       EEPROM_ASSERT( \

